src/native.rs

```diff
@@ -433,6 +438,20 @@ impl HasContext for Context {
         1 == status
     }
 
+    unsafe fn get_program_validate_status(&self, program: Self::Program) -> bool {
```

Comment:
I guess we could've exposed `get_program_parameter_bool`, `get_program_parameter_i32` (like you've added here), etc. instead of having these individual functions, but it's probably fine because there aren't too many anyway

Reply:
We should probably add `get_program_parameter_bool` and reuse these internally though (instead of having each of these repeating the `status == 1`), but we can do that in a follow-up PR